src/azure-cli/azure/cli/command_modules/iot/custom.py

```diff
@@ -441,8 +466,28 @@ def iot_hub_create(cmd, client, hub_name, resource_group_name, location=None,
                                         sku=sku,
                                         properties=properties,
                                         tags=tags)
-
-    return client.iot_hub_resource.create_or_update(resource_group_name, hub_name, hub_description)
+    hub_description.identity = _build_identity(identities) if identities else None
+    if bool(identity_role) ^ bool(identity_scopes):
+        raise CLIError('At least one scope (--scopes) and one role (--role) required for system-assigned managed identity role assignment')
```

Comment:
Please use the specific error type `RequiredArgumentMissingError` instead of `CLIError`.